### DISCUSSION TITLE
the rule priority issue for the block filter and Custom filter

### DISCUSSION THREAD

**Discussion by laughtLOOL** [AdguardTeam/AdGuardHome#4868] [Q&A] [answered]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not ask a question

### Operating system type

Linux, OpenWrt

### CPU architecture

ARMv7

### Installation

GitHub releases or script from README

### Setup

On a router, DHCP is handled by the router

### AdGuard Home version

v0.107.11

### Description

i not sure this problem which version happen, it such as 107.8 not show this question but this new version have that problem
it is a priority problem with the bolck filter and Custom filter, i not use the allow filter now but before i try to used. i just explain my question.
before the priority is bolck filter< Custom filter < allow filter in the not ^$inportant
and the bolck filter ^$important > Custom filter
and the bolck filter ^$important < Custom filter ^$important

this question is mistaken the bolck filter another freedom to use ^$important, and many another use that i not to change everyone,
 the WWW is test the list is have been run
 
![image](https://user-images.githubusercontent.com/83571492/186888160-7e07bc94-2eef-4462-b8cd-92dbf6193f6b.png)

and this is the problem, the Custom filter ^$important was not effect the bolck filter ^$important
i try to test the Custom filter ^$important with the bolck filter is effect
![image](https://user-images.githubusercontent.com/83571492/186888312-59273e01-38a5-4749-bce1-13f34e0f3c88.png)

and this is another list test

![image](https://user-images.githubusercontent.com/83571492/186888687-43a43f4e-dfd7-4097-9288-cef74a979d6a.png)

but use Custom filter cancel the block list was success
before non add list
![image](https://user-images.githubusercontent.com/83571492/186889786-ad458e16-ae16-4132-819f-0b2ab87b2194.png)
add the list have ^$important
![image](https://user-images.githubusercontent.com/83571492/186889598-97526e5c-32e6-40df-9dc8-beeff56580bc.png)
this is non ^$important
![image](https://user-images.githubusercontent.com/83571492/186890080-311cb662-1d3b-4d10-9e0c-dc1674723988.png)

so the question was show, || have the priority issue, make the Custom filter not effect
but the @@|| was no problem.

and in the same priority, @@|| high than ||.

**Comment by laughtLOOL** [comment]
i just give the rule link if want to test, not only that, many filter have same issue
https://raw.githubusercontent.com/2771936993/HG/main/hg.txt
![image](https://user-images.githubusercontent.com/83571492/187027389-dc488441-8d86-412b-912c-0175341bebda.png)

**Comment by laughtLOOL** [comment]
cant change the list was like that
@@||fengduxiaoshuo.com^$important
@@||pan.quark.cn^$important
@@||drive.quark.cn^$important
@@||pdds.ucweb.com^$important

but this block list can be change
||baichuan-sdk.taobao.com^$important
||baichuan-sdk.alicdn.com^$important
||oss.savor.com.cn^$important

**Comment by ameshkov** [comment, marked as answer]
The rules priority does not depend on what filter it comes from, it only depends on the modifiers.

Here's how it looks like: allowlist+$important > $important > allowlist > blocklist

Also, the more modifiers the rule has, the higher the priority is.

**Reply by laughtLOOL** [reply]
ok, clear that operate, and one thing
the rule can be write like that? ||oss.savor.com.cn^$important$important
to increased the priority, because now happen a host leak need to be solve, i cant remove the filter to solve the problem

**Reply by laughtLOOL** [reply]
what way can be increased the rule priority

**Reply by laughtLOOL** [reply]
and high than one important

**Reply by ameshkov** [reply]
Nope, that wouldn't help.

If you want to disable a rule existing in a filter list, you may want to use `$badfilter`.
Something like this:
`@@||oss.savor.com.cn^$important,badfilter`

**Reply by laughtLOOL** [reply]
Wow, useful
